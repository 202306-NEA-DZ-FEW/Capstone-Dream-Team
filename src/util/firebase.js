// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
