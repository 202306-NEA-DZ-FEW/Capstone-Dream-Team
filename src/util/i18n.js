import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
