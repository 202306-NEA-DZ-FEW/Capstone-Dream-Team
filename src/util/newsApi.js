export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
