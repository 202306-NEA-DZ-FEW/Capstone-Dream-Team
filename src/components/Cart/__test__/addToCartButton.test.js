import renderer from "react-test-renderer";

import { db } from "@/util/firebase";

import AddToCartButton from "../addToCartButton";

// Mock the Firestore collection and its methods
jest.mock("firebase/firestore", () => ({
    collection: jest.fn(),
}));

// Mock the Firestore instance
jest.mock("@/util/firebase", () => ({
    db: {
        collection: jest.fn(),
    },
}));

jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

jest.mock("next-i18next", () => ({
    useTranslation: jest.fn().mockReturnValue({ t: jest.fn() }),
}));

describe("AddToCartButton Component", () => {
    const mealObject = {
        quantity: "5",
        name: "baklawa",
        maxMeals: "20",
        price: "2",
        imageUrl: " ",
        // ... other properties
    };

    it("renders correctly", () => {
        // Mock Firestore behavior
        db.collection.mockReturnValue({
            where: jest.fn(() => ({ get: jest.fn() })),
            add: jest.fn(),
        });

        const tree = renderer
            .create(<AddToCartButton mealObject={mealObject} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
