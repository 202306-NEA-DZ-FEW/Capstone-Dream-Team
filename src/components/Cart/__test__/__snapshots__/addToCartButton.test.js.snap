// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton Component renders correctly 1`] = `
<div>
  <button
    className="w-60 h-11 left-0 top-0 absolute rounded-full text-center text-white text-xl font-bold uppercase leading-relaxed bg-orange-400 "
    onClick={[Function]}
  >
    Donate
  </button>
</div>
`;
